core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/controlBlock/ifStatement/IfStatementOpDesc.scala

```diff
@@ -0,0 +1,61 @@
+package edu.uci.ics.texera.workflow.operators.controlBlock.ifStatement
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonPropertyDescription}
+import com.kjetland.jackson.jsonSchema.annotations.JsonSchemaTitle
+import edu.uci.ics.amber.engine.architecture.deploysemantics.layer.OpExecInitInfo
+import edu.uci.ics.amber.engine.common.model.{PhysicalOp, SchemaPropagationFunc}
+import edu.uci.ics.amber.engine.common.model.tuple.Schema
+import edu.uci.ics.amber.engine.common.virtualidentity.{ExecutionIdentity, WorkflowIdentity}
+import edu.uci.ics.amber.engine.common.workflow.{InputPort, OutputPort, PortIdentity}
+import edu.uci.ics.texera.workflow.common.metadata.{OperatorGroupConstants, OperatorInfo}
+import edu.uci.ics.texera.workflow.common.operators.LogicalOp
+
+class IfStatementOpDesc extends LogicalOp {
+  @JsonProperty(required = true)
+  @JsonSchemaTitle("Condition State")
+  @JsonPropertyDescription("name of the state variable to evaluate")
+  var conditionName: String = _
+
+  override def getPhysicalOp(
+      workflowId: WorkflowIdentity,
+      executionId: ExecutionIdentity
+  ): PhysicalOp = {
+    PhysicalOp
+      .oneToOnePhysicalOp(
+        workflowId,
+        executionId,
+        operatorIdentifier,
+        OpExecInitInfo((_, _) => {
+          new IfStatementOpExec(conditionName)
+        })
+      )
+      .withInputPorts(operatorInfo.inputPorts)
+      .withOutputPorts(operatorInfo.outputPorts)
+      .withParallelizable(false)
+      .withPropagateSchema(
+        SchemaPropagationFunc(inputSchemas =>
+          operatorInfo.outputPorts
+            .map(_.id)
+            .map(id => id -> inputSchemas(operatorInfo.inputPorts.last.id))
+            .toMap
+        )
+      )
+  }
+
+  override def operatorInfo: OperatorInfo =
+    OperatorInfo(
+      "If Statement",
+      "If Statement",
+      OperatorGroupConstants.CONTROL_GROUP,
+      inputPorts = List(
+        InputPort(PortIdentity(), "Condition(State)"),
+        InputPort(PortIdentity(1), "Data", dependencies = List(PortIdentity()))
+      ),
+      outputPorts = List(OutputPort(PortIdentity(), "True"), OutputPort(PortIdentity(1), "False"))
```

Comment:
`"false"` and `"true"`

let port 0 be false 
port 1 be true.
this matches the boolean values of the port id.

Reply:
Switched.